File: wow-minmax-ui/src/app/modules/statistics/services/stats.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { CharacterStats } from '../model/CharacterStats';
import { RotationStats } from '../model/RotationStats';
import { SpecialAbilityStats } from '../model/SpecialAbilityStats';
import { SpellStats } from '../model/SpellStats';
import { TalentStats } from '../model/TalentStats';

@Injectable({
	providedIn: 'root'
})
export class StatsService {
	private readonly apiUrl = 'http://localhost:8080/api/v1/stats';

	constructor(private http: HttpClient) { }

	getSpellStats(characterId: string) {
		return this.http.get<SpellStats[]>(`${this.apiUrl}/${characterId}/spell`);
	}

	getCharacterStats(characterId: string) {
		return this.http.get<CharacterStats[]>(`${this.apiUrl}/${characterId}/character`);
	}

	getSpecialAbilities(characterId: string) {
		return this.http.get<SpecialAbilityStats[]>(`${this.apiUrl}/${characterId}/special`);
	}

	getRotationStats(characterId: string) {
		return this.http.get<RotationStats>(`${this.apiUrl}/${characterId}/rotation`);
	}

	getTalentStats(characterId: string) {
		return this.http.get<TalentStats[]>(`${this.apiUrl}/${characterId}/talent`);
	}
}

File: services/wow-evaluator/src/main/java/wow/evaluator/service/StatsService.java
package wow.evaluator.service;

import wow.character.model.build.Rotation;
import wow.commons.model.spell.AbilityId;
import wow.evaluator.model.*;

import java.util.List;

/**
 * User: POlszewski
 * Date: 2025-03-17
 */
public interface StatsService {
	List<SpellStats> getSpellStats(Player player, List<AbilityId> spells, boolean usesCombatRatings, double equivalentAmount);

	CharacterStats getCharacterStats(Player player, boolean worldBuffsAllowed);

	List<SpecialAbilityStats> getSpecialAbilityStats(Player player);

	RotationStats getRotationStats(Player player, Rotation rotation);

	List<TalentStats> getTalentStats(Player player);
}
